services/ymax-planner/test/tx-block-lower-bound.test.ts

```diff
@@ -0,0 +1,204 @@
+/**
+ * Tests for the functions that involve the use of `getTxBlockLowerBound` and `setTxBlockLowerBound`
+ */
+import test from 'ava';
+
+import { TxType } from '@aglocal/portfolio-contract/src/resolver/constants.js';
+import { createMockPendingTxData } from '@aglocal/portfolio-contract/tools/mocks.ts';
+import { getTxBlockLowerBound, setTxBlockLowerBound } from '../src/kv-store.ts';
+import { handlePendingTx } from '../src/pending-tx-manager.ts';
+import { EVENTS } from '../src/watchers/gmp-watcher.ts';
+import {
+  createMockGmpExecutionEvent,
+  createMockPendingTxOpts,
+} from './mocks.ts';
+import type { AccountId, CaipChainId } from '@agoric/orchestration';
+
+/**
+ * Sets up a test environment for testing transaction block lower bound functionality.
+ *
+ * Creates mock of all the utlities required by `handlePendingTx` inluding
+ * - EVM providers
+ * - Events associated with the providers
+ * - GMP tx data
+ * - logger utility
+ *
+ * @param latestBlock - The current latest block number in the mock evmProvider
+ * @param blockWithEvent - The block number where the mock event should be found
+ * @param txId - The transaction identifier to use for the pending transaction
+ * @param contractAddress - The destination contract address for the GMP transaction
+ */
+const setupEnvironment = ({
+  latestBlock,
+  blockWithEvent,
+  txId,
+  contractAddress,
+}: {
+  latestBlock: number;
+  blockWithEvent: number;
+  txId: `tx${number}`;
+  contractAddress: string;
+}) => {
+  const logs: string[] = [];
+  const mockLog = (...args: unknown[]) => logs.push(args.join(' '));
+
+  const chainId: CaipChainId = 'eip155:42161';
+  const destinationAddress: AccountId = `${chainId}:${contractAddress}`;
+  const gmpTx = createMockPendingTxData({
+    type: TxType.GMP,
+    destinationAddress,
+  });
+
+  const event = createMockGmpExecutionEvent(txId);
+  const opts = createMockPendingTxOpts(latestBlock, {
+    block: blockWithEvent,
+    event,
+  });
+  const mockProvider = opts.evmProviders[chainId];
+
+  const currentTimeMs = 1700000000; // 2023-11-14T22:13:20Z
+  const txTimestampMs = currentTimeMs - 10 * 1000; // 10 seconds ago
+  // Trigger block event to resolve waitForBlock
+  setTimeout(() => {
+    mockProvider.emit('block', latestBlock + 1);
+    latestBlock += 1;
+  }, 10);
+
+  return {
+    ctx: opts,
+    gmpTx,
+    mockLog,
+    logs,
+    txTimestampMs,
+  };
+};
+
+test('updates lower bound when a pending tx is being searched', async t => {
```

Comment:
```suggestion
test('deletes saved lower bounds when a pending tx is found', async t => {
```
We also seem to be missing a test demonstrating that lower bounds are ever updated; can you please add that one (or make this test cover both by adding relevant assertions)?